Utils/hdinsight-node-common/src/com/microsoft/azure/hdinsight/serverexplore/AddNewClusterCtrlProvider.java

```diff
@@ -56,14 +59,41 @@ public AddNewClusterCtrlProvider(@NotNull SettableControl<AddNewClusterModel> co
     }
 
     //format input string
-    private static String getClusterName(String userNameOrUrl) {
+    public static String getClusterName(String userNameOrUrl) {
         if (userNameOrUrl.startsWith(URL_PREFIX)) {
             return StringHelper.getClusterNameFromEndPoint(userNameOrUrl);
         } else {
             return userNameOrUrl;
         }
     }
 
+    public boolean isURLValid(@NotNull String url) {
```

Comment:
This method need to be `public` since it's used by `AddNewClusterForm.java` 

---
In reply to: [220066166](https://github.com/Microsoft/azure-tools-for-java/pull/2056#discussion_r220066166) [](ancestors = 220066166)